lihaoyi/Ammonite
# #460 - Read env proxy settings.

#460 

Ivy reads jvm system properties for proxy settings. This change looks for proxy info in environment variables, and set in system properties.

1. Added a utility object `ProxyFromEnv` that fetch current env proxy info.
2. Change only invades `IvyThing.scala` - above `ivy.resolve(md, options)`.
3. Unit test for `ProxyFromEnv` covers most common cases.
4. `System.setProperty()` in `predef.sc` will override envs. 

How did the integration test go(I had tough time compiling):

1. Use `ufw` to only allow proxy network.
2. Make `amm` run in its own jvm by  `set fork := true`.
3. put `interp.ivy.load...` in `/tmp/loadivy.sc`
4. run `test:run` with arg `"/tmp/loadivy.sc"` 

I enabled and disabled proxy resolution, it did what I expected.

Please check is there anything wrong or needed modification. Thanks.
